sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -249,16 +255,28 @@ public virtual async Task<Response> InsertOrReplaceAsync(IDictionary<string, obj
                 throw new ArgumentException(TableConstants.ExceptionMessages.MissingRowKey, nameof(entity));
             }
 
-            using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableClient)}.{nameof(InsertOrReplace)}");
+            using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableClient)}.{nameof(UpsertEntity)}");
             scope.Start();
             try
             {
-                return await _tableOperations.UpdateEntityAsync(_table,
+                if (mode == UpdateMode.Replace)
+                {
+                    return await _tableOperations.UpdateEntityAsync(_table,
+                                                            partitionKey as string,
+                                                            rowKey as string,
+                                                            tableEntityProperties: entity.ToOdataAnnotatedDictionary(),
+                                                            queryOptions: new QueryOptions() { Format = _format },
+                                                            cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
Nit - our general style convention is to either keep every param on the same line, or move every param on its own line with a single indent.  This would typically look like:

```C#
return await _tableOperations.UpdateEntityAsync(
    _table,
    partitionKey as string,
    rowKey as string,
    tableEntityProperties: entity.ToOdataAnnotatedDictionary(),
    queryOptions: new QueryOptions() { Format = _format },
    cancellationToken: cancellationToken).ConfigureAwait(false);
```